06week/higherOrder.js

```diff
@@ -3,15 +3,46 @@
 const assert = require('assert');
 
 function forEach(arr, callback) {
-  // Your code here
+  // Create a forEach() function that takes an array of items and a function that runs the function arr.length number of times.
+  const loopThrough = (arr, callback) => {
+  for (let i=0; i<arr.length; i++) {
+    console.log(arr[i]);
+
+    if (callback) {
+      console.log("callback present");
+
+      callback(arr[i]);
+    }
+  }
+}
 }
 
 function map(arr, callback) {
-  // Your code here
+  // Create a map() function that takes an array of items and a function that returns an array with each item manipulated by that function.
+  const mapEach = (arr, callback) => {
+  for (let i=0; i<arr.length; i++) {
+    console.log(arr[i]);
+
+    if (callback) {
+      console.log("callback present");
+
+      callback(arr[i]);
```

Comment:
Review the expected return of the map higher order function.  This function would not yield the expected return of map